third_party/move/move-model/src/constant_folder.rs

```diff
@@ -328,9 +328,17 @@ impl<'env> ConstantFolder<'env> {
                     O::Neq => val0
                         .equivalent(val1)
                         .map(|equivalence| V(id, Bool(!equivalence)).into_exp()),
-                    _ => self.constant_folding_error(id, |_| {
```

Comment:
The error can be raised when a constant is not evaluated. Thus modify the error here.